docs/rules/jsx-space-before-closing.md

```diff
@@ -1,8 +1,12 @@
-# Enforce spacing before closing bracket in JSX (react/jsx-space-before-closing)
+# Enforce spacing before closing bracket in JSX (`react/jsx-space-before-closing`)
 
-❌ This rule is deprecated. Please use the `"beforeSelfClosing"` option of the [jsx-tag-spacing](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-tag-spacing.md) rule instead.
+❌ This rule is deprecated. It was replaced by [jsx-tag-spacing](jsx-tag-spacing.md).
 
-🔧 This rule is automatically fixable using the `--fix` [flag](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix) on the command line.
+🔧 This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).
+
+<!-- end rule header -->
+
+Please use the `"beforeSelfClosing"` option of the [jsx-tag-spacing](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-tag-spacing.md) rule instead.
```

Comment:
it'd be better to keep this attached to the deprecation message if possible

Reply:
While the standard `meta.replacedBy` rule field allows the replacement rule to be indicated, there's no way for us to indicate a "replacement option" using the rule metadata. So currently there's no practical means for including this in the autogenerated content. Personally, it seems fine to me if some additional information about the deprecation comes later in the rule doc.

Reply:
hmm - standard wrt eslint itself?

Perhaps `replacedBy` could be extended to also support an object, with "rule" and "notes" or something?

Reply:
`meta.replacedBy` is defined as string array here:

* https://github.com/eslint/eslint/blob/4715787724a71494ba0bb0c5fe4639570bb6985b/lib/shared/types.js#L151
* https://github.com/typescript-eslint/typescript-eslint/blob/733b3598c17d3a712cf6f043115587f724dbe3ef/packages/utils/src/ts-eslint/Rule.ts#L74

So while I do think "deprecation message" fields are generally useful and could be used for the purpose you suggest, it would take a concerted effort to propose and update various definitions and tooling to handle objects like that. For now, I recommend we just expand on deprecations later in the rule doc.

Reply:
Sounds good, but it'd be great to pursue making that change :-)

Reply:
I could file an issue in ESLint suggesting it. I hope it wouldn't block this PR. 

Here's a potential API allowing deprecation messages about each specific replacement rule, or a general deprecation message (not specific to any replacement). Let me know if you like it.

```
replacedBy: ["replacement-rule-name"]
```

```
replacedBy: [{ name: "replacement-rule-name" }]
```

```
replacedBy: [{ name: "replacement-rule-name", message: "some deprecation message specific to this replacement" }]
```

```
replacedBy: [{ message: "general deprecation message" }]
```

Reply:
It wouldn’t block it. Your suggested schema seems awesome and backwards compatible.

Reply:
I finally wrote up a more detailed proposal for this:
* https://github.com/eslint/eslint/issues/18061

Reply:
Put together the RFC!
* https://github.com/eslint/rfcs/pull/116